### PR TITLE
Implement xDS circuit breaking interop test

### PR DESCRIPTION
**Pull request by voidzcy**
Implements xDS circuit breaking test. See go/grpc-xds-circuit-breaking-interop-tests.

For languages wanting to enable circuit breaking test, add `circuit_breaking` to the `--test_case` option and use the new server image `projects/grpc-testing/global/images/xds-test-server-3`.

/cc @menghanl @ericgribkoff @donnadionne

### CODE REVIEW

File: tools/run_tests/run_xds_tests.py [hunk 1]
```diff
@@ -1207,7 +1346,6 @@ def create_target_proxy(gcp, name):
         config = {
             'name': name,
             'url_map': gcp.url_map.url,
-            'validate_for_proxyless': True,
```
Comment (voidzcy) [inline tools/run_tests/run_xds_tests.py:1210]: TODO: should update the configuration check.

File: tools/run_tests/run_xds_tests.py [hunk 2]
```diff
@@ -311,6 +311,63 @@ def get_client_stats(num_rpcs, timeout_sec):
...
+        hosts = CLIENT_HOSTS
+    else:
+        hosts = ['localhost']
+    for host in hosts:
+        with grpc.insecure_channel('%s:%d' %
+                                   (host, args.stats_port)) as channel:
+            stub = test_pb2_grpc.XdsUpdateClientConfigureServiceStub(channel)
+            request = messages_pb2.ClientConfigureRequest()
+            for rpc_type in rpc_types:
+                if rpc_type not in ['empty_call', 'unary_call']:
```
Comment (ericgribkoff) [inline tools/run_tests/run_xds_tests.py:345]: Can the rpc_type parameter instead be a list of type `ClientConfigureRequest.RpcType`? Then we wouldn't need this check here or the `if rpc_type == 'empty_call'` conditional in constructing the `request` and `md` protobufs below.

File: tools/run_tests/run_xds_tests.py [hunk 3]
```diff
@@ -971,8 +1061,56 @@ def test_header_matching(gcp, original_backend_service, instance_group,
...
+        patch_backend_service(gcp, alternate_backend_service,
+                                [same_zone_instance_group],
+                                circuit_breakers={'maxRequests': max_requests})
+        wait_for_healthy_backends(gcp, alternate_backend_service,
+                                  same_zone_instance_group)
+        patch_url_map_backend_service(gcp, alternate_backend_service)
+        wait_until_all_rpcs_go_to_given_backends(alternate_backend_instances,
+                                                 _WAIT_FOR_URL_MAP_PATCH_SEC)
+        
+        # Make unary calls open.
```
Comment (ericgribkoff) [inline tools/run_tests/run_xds_tests.py:1087]: nit: s/open/keep-open/

File: tools/run_tests/run_xds_tests.py (hunk 1 above)
Comment (ericgribkoff) [reply tools/run_tests/run_xds_tests.py:1210]: Not sure I understand this - do we need a TODO in the code or will this be resolved before this PR is submitted?

File: tools/run_tests/run_xds_tests.py [hunk 4]
```diff
@@ -356,6 +413,39 @@ def wait_until_all_rpcs_go_to_given_backends(backends,
                                    num_rpcs,
                                    allow_failures=False)
 
+def wait_until_all_rpcs_fail(timeout_sec, num_rpcs):
```
Comment (ericgribkoff) [inline tools/run_tests/run_xds_tests.py:416]: Is this new method really needed? Can we just call `wait_until_all_rpcs_go_to_given_backends_or_fail([])` instead?

File: tools/run_tests/run_xds_tests.py (hunk 2 above)
Comment (voidzcy) [reply tools/run_tests/run_xds_tests.py]: Yeah, that's easier. Fix.

File: tools/run_tests/run_xds_tests.py (hunk 3 above)
Comment (voidzcy) [reply tools/run_tests/run_xds_tests.py]: Done.

File: tools/run_tests/run_xds_tests.py (hunk 1 above)
Comment (voidzcy) [reply tools/run_tests/run_xds_tests.py:1210]: Nvm, the config check of proxyless gRPC blocks us on configuring circuit breakers. I wasn't sure if this is something I need to do or someone else would do it. But it looks Richard is assigned for it. For now, we can just disable the config check.

**Comment by voidzcy** [comment]
Closing in favor of #24704.

**Pull request closed without merging**